search/controllers/api/__init__.py

```diff
@@ -144,37 +148,42 @@ def classic_query(params: MultiDict) \
 
     if raw_query:
         # migrate search_query -> query variable
-        params['query'] = raw_query
-        del params['search_query']
-
-        # pass to normal search, which will handle parsing
-        data, _, _ = search(params)
+        phrase = parse_classic_query(raw_query)
+        # TODO: add support for order, size, page_start
+        query = ClassicAPIQuery(phrase=phrase)
+        
+        # pass to search indexer, which will handle parsing
+        document_set = index.SearchSession.search(query)
+        data = {'results': document_set, 'query': query}
+        logger.debug('Got document set with %i results',
+                        len(document_set['results']))
 
     if id_list and not raw_query:
         # Process only id_lists.
         # Note lack of error handling to implicitly propogate any errors.
         # Classic API also errors if even one ID is malformed.
         papers = [paper(paper_id) for paper_id in id_list]
 
-        data, _, _ = zip(*papers)
-        results = [paper['results'] for paper in data]   # type: ignore
+        data, _, _ = zip(*papers) # type: ignore
+        results: List[Document] = [paper['results'] for paper in data] # type: ignore
         data = {
-            'results' : dict(results=results, metadata=dict()), # TODO: Aggregate search metadata
-            'query' : APIQuery() # TODO: Specify query
+            'results' : DocumentSet(results=results, metadata=dict()), # TODO: Aggregate search metadata
+            'query' : raw_query
         }
 
     elif id_list and raw_query:
         # Filter results based on id_list
-        results = [paper for paper in data['results']['results']
+        results: List[Document] = [paper for paper in data['results'].results # type: ignore
                        if paper.paper_id in id_list or paper.paper_id_v in id_list]
         data = {
-            'results' : dict(results=results, metadata=dict()), # TODO: Aggregate search metadata
+            'results' : DocumentSet(results=results, metadata=dict()), # TODO: Aggregate search metadata
             'query' : APIQuery() # TODO: Specify query
```

Comment:
shouldn't this be ``'query': raw_query`` as above?

Reply:
fixed the `data['query']` to always use a `Query` object instead,